Rectangle.cpp

```diff
@@ -1,15 +1,48 @@
 #ifndef Rectangle_cpp
 #define Rectangle_cpp
+
 #include <iostream>
-#include "Draw.cpp"
+#include "Shape.cpp"
+#include "Area.cpp"
+#include "Input.cpp"
+
 using namespace std;
 
-class Rectangle : public Draw
+class Rectangle : public Shape, public Input
 {
-public:
-    void drawShape() override
-    {
-        cout << "rectangle" << endl;
-    }
+    private:
+        double length = input();
+        double width = input();
+
+    public:
+        void drawShape() override
+        {
+            cout << "rectangle" << endl;
+        }
+
+        double input() override
+        {
+            if (length>0)
```

Comment:
It looks like `width` will never be filled unless length is zero. This is a bug, to calculate the area of a rectangle we need both length and width.